crates/factor-key-value-spin/src/lib.rs

```diff
@@ -22,14 +26,20 @@ impl SpinKeyValueStore {
     }
 }
 
+/// Runtime configuration for the SQLite key-value store.
 #[derive(Deserialize, Serialize)]
 pub struct SpinKeyValueRuntimeConfig {
+    /// The path to the SQLite database file.
     path: Option<PathBuf>,
 }
 
 impl SpinKeyValueRuntimeConfig {
+    /// The default filename for the SQLite database.
     const DEFAULT_SPIN_STORE_FILENAME: &'static str = "sqlite_key_value.db";
 
+    /// Create a new runtime configuration with the given state directory.
+    /// If the state directory is None, the database is in-memory.
+    /// If the state directory is Some, the database is stored in a file in the state directory.
     pub fn default(state_dir: Option<PathBuf>) -> Self {
```

Comment:
Instead of calling this `state_dir`, perhaps we can call it `default_database_dir`? I think this is a bit more description of what this is actually being used for. The fact that in Spin CLI it's the state directory is something that the caller knows but that this implementation doesn't necessarily need to care about. We do [this for sqlite](https://github.com/fermyon/spin/blob/7c6faed98a55a8b5ef6a73f3014c8158a39fe251/crates/factor-sqlite/src/runtime_config/spin.rs#L37).